lib/assert.js

```diff
@@ -837,22 +902,51 @@ function expectsNoError(stackStartFn, actual, error, message) {
   throw actual;
 }
 
+/**
+ * Expects the function `promiseFn` to throw an error.
+ * @param {Function} promiseFn
```

Comment:
```suggestion
 * @param {() => any} promiseFn
```

Slight differences here, mostly just listing that expected arguments length is 0